taichi/backends/opengl/opengl_api.cpp

```diff
@@ -521,7 +526,7 @@ GLSLLaunchGuard::GLSLLaunchGuard(GLSLLauncherImpl *impl,
   for (int i = 0; i < impl->ssbo.size(); i++) {
     if (!iov[i].size)
       continue;
-    impl->ssbo[i].bind_index(i + 1);
+    impl->ssbo[i].bind_index(i + 2);                    // skip root and gtmp
```

Comment:
Out of curiosity, why can we skip `gtmp` now?

Reply:
`impl->ssbo`'s here are bound at runtime, that's why we need this guard.
`impl->gtmp_ssbo` and `impl->root_ssbo` are already allocated&bound compile-time.
Used to we only have `root` bound at compile-time, now we move `gtmp` to compile-time too. That's because gtmp is big enough (1MB), so we don't need update it at runtime now, therefore save overhead (see TPRT result in PR desc).

Reply:
Ah i see, thx